java/team25core/ObjectDetectionTask.java

```diff
@@ -0,0 +1,250 @@
+package team25core;
+
+import com.qualcomm.robotcore.hardware.HardwareMap;
+import com.qualcomm.robotcore.util.ElapsedTime;
+import com.vuforia.PIXEL_FORMAT;
+import com.vuforia.Vuforia;
+
+import org.firstinspires.ftc.robotcore.external.ClassFactory;
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
+import org.firstinspires.ftc.robotcore.external.navigation.VuforiaLocalizer;
+import org.firstinspires.ftc.robotcore.external.tfod.Recognition;
+import org.firstinspires.ftc.robotcore.external.tfod.TFObjectDetector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import team25core.Robot;
+import team25core.RobotEvent;
+import team25core.RobotTask;
+import team25core.VuforiaConstants;
+
+
+public class ObjectDetectionTask extends RobotTask {
+
+    public enum EventKind {
+        OBJECTS_DETECTED,
+    }
+
+    protected ElapsedTime timer;
+
+    public class ObjectDetectionEvent extends RobotEvent {
+
+        public EventKind kind;
+        public List<Recognition> objects;
+
+        //this is constructor for object detection event
+        public ObjectDetectionEvent(RobotTask task, EventKind kind, List<Recognition> m)
+        {
+            super(task);
+            this.kind = kind;
+            this.objects = new ArrayList<>(m.size());
+            this.objects.addAll(m);
+        }
+
+        public String toString()
+        {
+            return kind.toString();
+        }
+    }
+
+    private VuforiaLocalizer vuforia;
+    private Telemetry telemetry;
+    private TFObjectDetector tfod;
+
+    public static final String LABEL_OBJECT1 = "Element";
+    private static final String TFOD_MODEL_ASSET = "QTFreightFrenzy.tflite";
+    private int rateLimitMs;
+    private DetectionKind detectionKind;
+    private String cameraName;
+
+    public enum DetectionKind {
+        EVERYTHING, //this may go away
+        OBJECT1_DETECTED,
+        OBJECT2_DETECTED,
+        LARGEST_SKY_STONE_DETECTED, //this may go away
+        UNKNOWN_DETECTED,
+    }
+     public enum ObjectKind {
+            OBJECT2_KIND,
+            OBJECT1_KIND,
+            UNKNOWN_KIND,
+     };
+
+
+    //for phone camera constructor
+    public ObjectDetectionTask(Robot robot)
+    {
+        super(robot);
+
+        rateLimitMs = 0;
+        detectionKind = DetectionKind.EVERYTHING;
+    }
+    //for webcamera construtor
+    public ObjectDetectionTask(Robot robot, String cameraName)
+    {
+        super(robot);
+        rateLimitMs = 0;
+        detectionKind = DetectionKind.EVERYTHING;
+        this.cameraName = cameraName;
+    }
+
+    private void initVuforia(HardwareMap hardwareMap) {
+        /*
+         * Configure Vuforia by creating a Parameter object, and passing it to the Vuforia engine.
+         */
+        //new=your own copy of vuforia parameters
```

Comment:
"new" means create a new instance of the class which allocates memory for the new instance distinct and separate from any other instances.   Somewhat pedantic in that this gives you a "copy" but the terminology is just ever so slightly off.